tests/ct_connection.c

```diff
@@ -66,22 +69,87 @@ void test_password_missing() {
     redisClusterFree(cc);
 }
 
+// Connect and handle two clusters simultaneously
+void test_multicluster() {
+    int status;
+    redisReply *reply;
+
+    // Connect to first cluster
+    redisClusterContext *cc1 = redisClusterContextInit();
+    assert(cc1);
+    redisClusterSetOptionAddNodes(cc1, CLUSTER_NODE);
+    status = redisClusterConnect2(cc1);
+    ASSERT_MSG(status == REDIS_OK, cc1->errstr);
+
+    // Connect to second cluster
+    redisClusterContext *cc2 = redisClusterContextInit();
+    assert(cc2);
+    redisClusterSetOptionAddNodes(cc2, CLUSTER_NODE_WITH_PASSWORD);
+    redisClusterSetOptionPassword(cc2, CLUSTER_PASSWORD);
+    status = redisClusterConnect2(cc2);
+    ASSERT_MSG(status == REDIS_OK, cc2->errstr);
+
+    // Set keys differently in clusters
+    reply = redisClusterCommand(cc1, "SET key Hello1");
+    CHECK_REPLY_OK(cc1, reply);
+    freeReplyObject(reply);
+
+    reply = redisClusterCommand(cc2, "SET key Hello2");
+    CHECK_REPLY_OK(cc2, reply);
+    freeReplyObject(reply);
+
+    // Verify keys in clusters
+    reply = redisClusterCommand(cc1, "GET key");
+    CHECK_REPLY_STR(cc1, reply, "Hello1");
+    freeReplyObject(reply);
+
+    reply = redisClusterCommand(cc2, "GET key");
+    CHECK_REPLY_STR(cc2, reply, "Hello2");
+    freeReplyObject(reply);
+
+    // Disconnect from first cluster
+    redisClusterFree(cc1);
+
+    // Verify that key is still accessible in connected cluster
+    reply = redisClusterCommand(cc2, "GET key");
+    CHECK_REPLY_STR(cc2, reply, "Hello2");
+    freeReplyObject(reply);
+
+    redisClusterFree(cc2);
+}
+
 //------------------------------------------------------------------------------
 // Async API
 //------------------------------------------------------------------------------
-
+typedef struct ExpectedResult {
+    int type;
+    char *str;
+    bool disconnect;
+    bool noreply;
+    char *errstr;
+} ExpectedResult;
+
+// Callback for Redis connects and disconnects
 void callbackExpectOk(const redisAsyncContext *ac, int status) {
     UNUSED(ac);
     assert(status == REDIS_OK);
 }
 
+// Callback for async commands, verifies the redisReply
 void commandCallback(redisClusterAsyncContext *cc, void *r, void *privdata) {
-    UNUSED(r);
-    UNUSED(privdata);
     redisReply *reply = (redisReply *)r;
-    assert(reply != NULL);
-    assert(strcmp(reply->str, "OK") == 0);
-    redisClusterAsyncDisconnect(cc);
+    ExpectedResult *expect = (ExpectedResult *)privdata;
+    if (expect->noreply) {
+        assert(reply == NULL);
+        assert(strcmp(cc->errstr, expect->errstr) == 0);
+    } else {
+        assert(reply != NULL);
+        assert(reply->type == expect->type);
+        if (reply->type != REDIS_REPLY_INTEGER)
+            assert(strcmp(reply->str, expect->str) == 0);
```

Comment:
Not integer doesn't imply it's a string reply, does it? Can't it be an array or anything else...?

Reply:
Yes, this looked a bit awkward now. Changed it to only check for types that uses the `str` element in the reply